.travis.yml

```diff
@@ -0,0 +1,140 @@
+    #- libgit2-dev
+    # markdown
+    - libmarkdown2-dev
+    # markdown/webhelper
+    - libwebkitgtk-3.0-dev
+    # multiterm
+    - valac
+    # pretty-printer
+    - libxml2-dev
+    # spellcheck
+    - libenchant-dev
+
+    # for libgit2 build
+    - cmake
+
+before_install:
+  # build Geany
+  - TEMPDIR=$(mktemp -d)
+  - git clone git://github.com/geany/geany.git "$TEMPDIR/geany"
+  - geany_commit=$(cd "$TEMPDIR/geany" && git rev-parse HEAD)
+  - gtkver=gtk2
+  - test "x$GTK3" = xno || gtkver=gtk3
+  - cachedir="$HOME/geany/$gtkver/"
+  - cache_id=$(cat "$cachedir/commit-id" 2>/dev/null || echo "none")
+  # check if the cache is up-to-date, and either use it, or (re-)create it
+  - >
+    if test "$geany_commit" = "$cache_id"; then
+      echo "Cache '$cachedir' looks good, using it (ID: $cache_id).";
+    else
+      echo "Cache '$cachedir' is missing, outdated or invalid, dropping it.";
+      echo "  Cache ID: $cache_id";
+      echo "  Geany ID: $geany_commit";
+      rm "$cachedir" -rf || exit 1;
+      ( cd "$TEMPDIR/geany"                                   &&
+        NOCONFIGURE=1 ./autogen.sh                            &&
+        ./configure --prefix="$cachedir" --enable-gtk3=$GTK3  &&
+        make -j2                                              &&
+        make install || exit 1;
+        git rev-parse HEAD > "$cachedir/commit-id" || exit 2;
+      ) || exit $?;
+    fi
+  - test -z "$TEMPDIR" || rm -rf "$TEMPDIR"
+  - export PKG_CONFIG_PATH="$cachedir/lib/pkgconfig:$PKG_CONFIG_PATH"
+  - export LD_LIBRARY_PATH="$cachedir/lib:$LD_LIBRARY_PATH"
+
+  # build libgit2
+  - TEMPDIR=$(mktemp -d)
+  - libgit2dir="libgit2-$LIBGIT2_VERSION"
+  - >
+    if test -f "$HOME/$libgit2dir/lib/pkgconfig/libgit2.pc"; then
+      echo "Using pre-built libgit2 v$LIBGIT2_VERSION.";
+    else
+      ( cd $TEMPDIR || exit 1;
+        wget -q "https://github.com/libgit2/libgit2/archive/v$LIBGIT2_VERSION.tar.gz" -O - |
+          tar -xz "$libgit2dir" || exit 2;
+        cd "$libgit2dir" || exit 3;
+        cmake .                                       \
+          -DCMAKE_INSTALL_PREFIX="$HOME/$libgit2dir"  \
+          -DCMAKE_RULE_MESSAGES=OFF                   \
+          -DBUILD_CLAR=OFF                            \
+          -DBUILD_SHARED_LIBS=ON                      \
+          || exit 4;
+        cmake --build . --target install -- -j2 || exit 5;
+      ) || exit $?;
+    fi
+  - test -z "$TEMPDIR" || rm -rf "$TEMPDIR"
+  - export PKG_CONFIG_PATH="$HOME/$libgit2dir/lib/pkgconfig:$PKG_CONFIG_PATH"
+  - export LD_LIBRARY_PATH="$HOME/$libgit2dir/lib:$LD_LIBRARY_PATH"
+
+before_script:
+  # prepare for GP
+  - export CFLAGS="-g -O2 -Werror=pointer-arith -Werror=aggregate-return -Werror=implicit-function-declaration"
+
+script:
+  - >
+    NOCONFIGURE=1 ./autogen.sh  &&
+    mkdir _build                &&
+    cd _build                   &&
+    ../configure                &&
+    make -j2                    &&
```

Comment:
Do we really want to build with j2? On Travis I don't see much we are winning but error output might be harder to read.